### PR TITLE
stm32l4r9ai-disco: set the heap size with 640KB

### PR DESCRIPTION
**Pull request by sunghan-chang**

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [139248133ce92ed28eb4c518e0b30c00240078dc] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [139248133ce92ed28eb4c518e0b30c00240078dc] - Code Rule Check (C++) OK.

File: os/arch/arm/src/stm32l4/stm32l4_allocateheap.c [hunk 1]
```diff
@@ -323,63 +395,30 @@ void up_allocate_kheap(FAR void **heap_start, size_t *heap_size)
...
-
-  /* Colorize the heap for debug */
-
-  up_heap_color((FAR void *)SRAM2_START, SRAM2_END-SRAM2_START);
-
-  /* Add the SRAM2 user heap region. */
-
-  kumm_addregion((FAR void *)SRAM2_START, SRAM2_END-SRAM2_START);
-
+  heap_size = up_calculte_heapsize((void *)SRAM2_START, SRAM2_END);
```
Comment (jeongchanKim) [inline os/arch/arm/src/stm32l4/stm32l4_allocateheap.c]: up_calculte_heapsize -> up_calculate_heapsize

**Comment by seinfra** [comment]
Target : [8ddbe77e01f6be99fe1fabaa628feb6eceb9dc46] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [8ddbe77e01f6be99fe1fabaa628feb6eceb9dc46] - Code Rule Check (C++) OK.

File: os/arch/arm/src/stm32l4/stm32l4_allocateheap.c [hunk 2]
```diff
@@ -224,46 +260,68 @@ void up_allocate_heap(FAR void **heap_start, size_t *heap_size)
...
+
+    board_led_on(LED_HEAPALLOCATE);
+    *heap_start = (FAR void *)ubase;
+    *heap_size  = usize;
 
-  up_heap_color((FAR void *)ubase, usize);
+    /* Colorize the heap for debug */
 
-  /* Allow user-mode access to the user heap memory */
+    up_heap_color((FAR void *)ubase, usize);
```
Comment (sangwon03) [inline os/arch/arm/src/stm32l4/stm32l4_allocateheap.c]: why don't you use 'heap_start' instead of  '(FAR void  *)usbase'?

Comment (sunghan-chang) [reply os/arch/arm/src/stm32l4/stm32l4_allocateheap.c]: In protected build case, it starts from base address of user space. That's why it named as ubase.

Comment (sunghan-chang) [reply os/arch/arm/src/stm32l4/stm32l4_allocateheap.c]: And it's a chip-specific file, we had not better modify it by ourself if it is not wrong.

**Pull request merged**